mesonbuild/interpreter/interpreter.py

```diff
@@ -2674,23 +2673,13 @@ def func_configure_file(self, node: mparser.BaseNode, args: T.List[TYPE_var],
                                               install_tag=install_tag, data_type='configure'))
         return mesonlib.File.from_built_file(self.subdir, output)
 
-    def extract_incdirs(self, kwargs, key: str = 'include_directories'):
-        prospectives = extract_as_list(kwargs, key)
-        if key == 'include_directories':
-            for i in prospectives:
-                if isinstance(i, str):
-                    FeatureNew.single_use('include_directories kwarg of type string', '0.50.0', self.subproject,
-                                          f'Use include_directories({i!r}) instead', location=self.current_node)
-                    break
-
-        result = []
```

Comment:
While rebasing, you dropped this code, which recently added a warning if people used `include_directories: 'somestring'` (but isn't needed for d_import_dirs).

Reply:
I've added some commits in the latest version to emit the same FeatureNew warning.